src/strands/models/openai.py

```diff
@@ -532,6 +564,11 @@ async def stream(
             if event and hasattr(event, "usage") and event.usage:
                 yield self.format_chunk({"chunk_type": "metadata", "data": event.usage})
 
+        finally:
+            # Only close the client if we created it (not injected)
+            if self._injected_client is None and hasattr(client_to_use, "close"):
```

Comment:
In what case would `close` not be present?  I feel like this check shouldn't be needed?